quickwit-index-config/src/default_index_config/default_config.rs

```diff
@@ -43,20 +43,36 @@ pub struct DefaultIndexConfigBuilder {
     store_source: bool,
     default_search_fields: Vec<String>,
     timestamp_field: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sort_by: Option<SortByConfig>,
     field_mappings: Vec<FieldMappingEntry>,
     tag_fields: Vec<String>,
 }
 
+#[derive(Default, Serialize, Deserialize, Clone)]
+struct SortByConfig {
+    /// Sort field name in the index schema.
+    pub field_name: String,
+    pub order: SortOrder,
+}
+
+impl From<SortByConfig> for SortBy {
+    fn from(sort_by_config: SortByConfig) -> Self {
+        SortBy::SortByFastField {
+            field_name: sort_by_config.field_name,
+            order: sort_by_config.order,
+        }
+    }
+}
```

Comment:
looks like SortBy::SortByFastField is the same as SortByConfig, so we could reuse it there
```
#[derive(Clone, Debug)]
pub enum SortBy {
    /// Sort by a specific field.
    SortByFastField (SortByFastFieldConfig),
    /// Sort by DocId
    DocId,
}
```

Reply:
I prefer the way it is today.

Reply:
@PSeitz this is the way :D 
I was hesitating and have found that it was slightly hurting readability and I like keeping away all the bloated config structs away from the core as much as possible. I would keep it as it is this for now.